tests/lib/config.js

```diff
@@ -761,6 +761,19 @@ describe("Config", function() {
             }, /Referenced from:.*?error\.json/);
         });
 
+        // Keep order with the last array element taking highest precedence
+        it("should make the last element in an array take the highest precedence", function() {
```

Comment:
Hmmm.. I see that the unittest is passing, even though I would expect it to fail and value of `comma-dangle` to be 1... Strange, reverse should've merged in `.eslintrc2` first, and then override it with `.eslintrc1`. So my comment above might not be valid, but I still think it's much better to use `reduceRight` in that case.